.github/actions/notify-pull-request/action.yml

```diff
@@ -22,12 +22,19 @@ runs:
       if: ${{ inputs.remove-previous-comment == 'true' }}
       with:
         script: |
+          const pr = context.payload.pull_request ?? context.payload.issue;
+
+          if (!pr) {
+            console.error("This action can only be run on pull requests");
+            return;
+          }
```

Comment:
this will return with exit code 0, which might cause this step to succeed when it should fail. I think a better approach is for the calling job to skip this step in non-PR contexts like:


```yaml
if: ${{ always() && github.event_name == 'pull_request' }
```

Reply:
I did the same way as the `- name: Notify pull-request` step was done.
I can adjust but only this step to avoid any unexpected sidefects of `Notify pull-request`

Reply:
Adjusted to 

```
if: ${{ inputs.remove-previous-comment == 'true' && github.event_name == 'pull_request' }}
```

Not sure if we need `always()` here. It is the first step of the action.

Reply:
Actually let's add it `always()` as the safe point to both actions